reactivesocket-core/src/main/java/io/reactivesocket/util/PayloadImpl.java

```diff
@@ -25,37 +25,35 @@
 /**
  * An implementation of {@link Payload}. This implementation is <b>not</b> thread-safe, and hence any method can not be
  * invoked concurrently.
- *
- * <h2>Reusability</h2>
- *
- * By default, an instance is reusable, i.e. everytime {@link #getData()} or {@link #getMetadata()} is invoked, the
- * position of the returned buffer is reset to the start of data in the buffer. For creating an instance for single-use,
- * {@link #PayloadImpl(ByteBuffer, ByteBuffer, boolean)} must be used.
  */
 public class PayloadImpl implements Payload {
 
-    public static final Payload EMPTY = new PayloadImpl(Frame.NULL_BYTEBUFFER, Frame.NULL_BYTEBUFFER);
+    public static final PayloadImpl EMPTY = new PayloadImpl(Frame.NULL_BYTEBUFFER, Frame.NULL_BYTEBUFFER, false);
 
     private final ByteBuffer data;
+    private final ByteBuffer metadata;
     private final int dataStartPosition;
     private final int metadataStartPosition;
     private final boolean reusable;
-    private final ByteBuffer metadata;
+
+    public PayloadImpl(Frame frame) {
+        this(frame.getData(), frame.getMetadata());
+    }
 
     public PayloadImpl(String data) {
-        this(data, (String) null);
+        this(data, Charset.defaultCharset());
     }
 
     public PayloadImpl(String data, String metadata) {
-        this(fromString(data), fromString(metadata));
+        this(data, Charset.defaultCharset(), metadata, Charset.defaultCharset());
```

Comment:
Does defaultCharset() make sense for a network centric application? Should we just be more UTF_* opinionated instead?